src/Facebook/Facebook.php

```diff
@@ -586,4 +589,72 @@ public function videoToUpload($pathToFile)
     {
         return new FacebookVideo($pathToFile);
     }
+
+    /**
+     * Upload video
+     *
+     * @param int $target
+     * @param string $pathToFile
+     * @param string $accessToken
+     * @param int $maxTransferTries
+     *
+     * @return array
+     *
+     * @throws FacebookSDKException
+     */
+    public function uploadVideo($target, $pathToFile, $accessToken, $maxTransferTries = 5)
+    {
+        $uploader = $this->getResumableUploader($accessToken, $maxTransferTries);
+        $endpoint = '/'.$target.'/videos';
+        $chunk = $uploader->start($endpoint, $pathToFile);
+
+        do {
+            $chunk = $uploader->transfer($endpoint, $chunk);
+        } while(!$chunk->isLastChunk());
+
+        return [
+            'videoId' => $chunk->getVideoId(),
+            'success' => $uploader->finish($endpoint, $chunk->getUploadSessionId()),
+        ];
+    }
+
+    /**
+     * Instantiates a new video uploader
+     *
+     * @param string $accessToken
+     * @param int $maxTransferTries
+     *
+     * @return FacebookResumableUploader
+     */
+    protected function getResumableUploader($accessToken, $maxTransferTries = 5)
+    {
+        return new FacebookResumableUploader($this->app, $this->client, $accessToken, $maxTransferTries);
```

Comment:
@yguedidi I can see your point. There are a number of features of the SDK that make HTTP calls and they are not always so obvious to the end-user that that's what happening in the background (like `getAccessToken()` for example). That's the main reason I suggested the name change, but again, I'm cool with keeping it as-is. :)